web/services/counter_service.py

```
@@ -0,0 +1,101 @@
+
+    @staticmethod
+    def get_recent_activities(limit=10):
+        """Get the most recent user activities for display."""
+        return UserActivity.objects.all()[:limit]
+
+
+# Signal receivers to update counters and record activities
+
+
+@receiver(post_save, sender=Enrollment)
+def enrollment_created(sender, instance, created, **kwargs):
```

Ruff (0.8.2)

Missing return type annotation for public function `enrollment_created`

Add return type annotation: `None`

(ANN201)